### PR TITLE
fix: InputNumber ime mode bug

### LINKED ISSUE #21199: Input numbers using the iPhone's Japanese keyboard
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/input-number/](https://ant.design/components/input-number/)

### Steps to reproduce
1. This could be reproduced on any browser on iPhone
2. Switch keyboard to Japanese
3. Type in a random range of numeric characters on Antd InputNumber

### What is expected?
It doubles some character we type in, sometimes it double not just one but a group of characters.

### What is actually happening?
Unexpected display behavior when type in numbers

| Environment | Info |
|---|---|
| antd | 3.26.5 |
| React | 16.12 |
| System | iOS |
| Browser | any |

---
This could also be reproduced on Windows. Step to reproduce:
1. Install Google Japanese Input  (link: https://tools.google.com/dlpage/japaneseinput/eula.html )
2. Go to https://ant.design/components/input-number/
3. Switch keyboard input to Google Japanese Input
4. Type in a random range of numeric characters on Antd InputNumber

### PR DESCRIPTION
**Pull request by afc163**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #21199

### 💡 Background and solution

https://github.com/ant-design/ant-design/issues/21199#issuecomment-582269009

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix InputNumber bug of using iOS Japanese keyboard. |
| 🇨🇳 Chinese | 修复 InputNumber 在 iOS 日文输入法下的问题。  |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
TypeScript error: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/37660

File: package.json [hunk 1]
```diff
@@ -152,7 +152,7 @@
     "@types/lodash": "^4.14.139",
     "@types/prop-types": "^15.7.1",
     "@types/raf": "^3.4.0",
-    "@types/react": "^16.9.0",
+    "@types/react": "16.9.20",
```
Comment (afc163) [inline package.json]: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42398

**Comment by eps1lon** [comment]
@afc163 You can't pass an actual function component to `React.forwardRef`.

In https://github.com/ant-design/ant-design/blob/1394e0a651e6d17416a09537a82df1a35437221c/components/form/Form.tsx#L28
```patch
-const InternalForm: React.FC<FormProps> = (props, ref) => {
-const InternalForm: React.ForwardRefRenderFunction<unknown, FormProps> = (props, ref) => {
```
should work.

**Comment by afc163** [comment]
@eps1lon Thx! #21483

File: components/input-number/index.tsx [hunk 2]
```diff
@@ -74,6 +74,7 @@ export default class InputNumber extends React.Component<InputNumberProps, any>
           return (
             <RcInputNumber
               ref={this.saveInputNumber}
+              type="tel"
```
Comment (zombieJ) [inline components/input-number/index.tsx]: Why set `tel` here?

Comment (zombieJ) [reply components/input-number/index.tsx]: https://github.com/ant-design/ant-design/issues/21199#issuecomment-582269009

Comment (zombieJ) [reply components/input-number/index.tsx]: Hmm...this may affect accessibility.

Comment (afc163) [reply components/input-number/index.tsx]: Some how?

Comment (zombieJ) [reply components/input-number/index.tsx]: Will tread input as a tel input.

**Comment by yoyo837** [comment]
#21199 咋没被close掉。

**Pull request closed without merging**